templates/Segment_Image_Data.ipynb

```diff
@@ -281,12 +281,14 @@
     "segmentation_labels_cell = load_utils.load_imgs_from_dir(data_dir=deepcell_output_dir,\n",
     "                                                    xr_dim_name='compartments',\n",
     "                                                    xr_channel_names=['whole_cell'],\n",
+    "                                                    files=[fov + '_feature_0.tif' for fov in fovs],\n",
     "                                                    delimiter='_feature_0',\n",
     "                                                    force_ints=True)\n",
     "\n",
     "segmentation_labels_nuc = load_utils.load_imgs_from_dir(data_dir=deepcell_output_dir,\n",
     "                                                    xr_dim_name='compartments',\n",
     "                                                    xr_channel_names=['nuclear'],\n",
+    "                                                    files=[fov + '_feature_1.tif' for fov in fovs],\n",
```

Comment:
We need to address the underlying issue with the load_imgs function, otherwise this is just going to happen again. 

No error message was thrown when the delimiter didn't match the file name. Was it you or @ackagel who added the `delimiter_optional` flag to the `extract_delimited_names` function? What was the reasoning behind that?

Reply:
Sounds my (apparently not so handi)-work.  I'll look into that, quickly rework the functionality, and rewrite/reimplement the test functions which should've caught this failure.

Reply:
@alex-l-kong I'll work on a PR for that and merge it into this branch once it's approved.  Sounds good?